monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/delegates/TemplatesDelegate.java

```diff
@@ -20,148 +20,27 @@
-                                                                     mgtApiUrl, accessToken);
-                                                                     
-                AtomicBoolean isRecordExist = new AtomicBoolean(false);
-                artifacts.stream().filter(o -> (o.getName()).equals(templateName)).forEach(
-                        o -> {
-                            o.getNodes().add(artifactDetails);
-                            isRecordExist.set(true);
-                        });
-                if (!isRecordExist.get()) {
-                    ArtifactsInner artifactsInner = new ArtifactsInner();
-                    artifactsInner.setName(templateName);
-                    List<ArtifactDetails> artifactDetailsList = new ArrayList<>();
-                    artifactDetailsList.add(artifactDetails);
-                    artifactsInner.setNodes(artifactDetailsList);
-                    artifactsInner.setType(templateType);
-                    artifacts.add(artifactsInner);
-                }
-            }
-        }
-
-        //ordering   
-        Comparator<ArtifactsInner> comparatorObject;
-        switch (orderBy.toLowerCase()) {
-            //for any other ordering options
-            default: comparatorObject = Comparator.comparing(ArtifactsInner::getNameIgnoreCase); break;
-        }
-        if ("desc".equalsIgnoreCase(order)) {
-            Collections.sort(artifacts, comparatorObject.reversed());
-        } else {
-            Collections.sort(artifacts, comparatorObject);
-        }
-        return artifacts;
-    }
-
-    private static ArtifactDetails getTemplateDetails(String groupId, String nodeId, String type, String name,
-            String mgtApiUrl, String accessToken)
-        throws ManagementApiException {
-        
-        JsonObject details = getArtifactDetails(groupId, nodeId, mgtApiUrl, type, name, accessToken);
-        ArtifactDetails artifactDetails = new ArtifactDetails();
-        artifactDetails.setNodeId(nodeId);
-        artifactDetails.setDetails(details.toString());
-        return artifactDetails;
-    }
-
-    private static JsonObject getArtifactDetails(String groupId, String nodeId, String mgtApiUrl, String templateType,
-                                                String templateName, String accessToken) throws ManagementApiException {
-        
-        String getArtifactDetailsUrl = mgtApiUrl.concat(Constants.TEMPLATES).concat("?type=").
-            concat(templateType).concat("&name=").concat(templateName);
-        CloseableHttpResponse artifactDetails = Utils.doGet(groupId, nodeId, accessToken,
-                                                            getArtifactDetailsUrl);
-        JsonObject jsonResponse = HttpUtils.getJsonResponse(artifactDetails);
-        jsonResponse.remove("configuration");
-        return jsonResponse;
+        DelegatesUtil.logDebugLogs(Constants.TEMPLATES, groupId, lowerLimit, upperLimit, order, orderBy, isUpdate);
```

Comment:
Could you test the pagination for templates in Dashboard?

Reply:
Done. Thank you.